tianshou/data/batch.py

```diff
@@ -194,12 +203,12 @@ def __getstate__(self) -> dict:
             state[k] = v
         return state
 
-    def __setstate__(self, state) -> None:
+    def __setstate__(self, state: Dict[str, Any]) -> None:
         """Unpickling interface. At this point, self is an empty Batch instance
         that has not been initialized, so it can safely be initialized by the
         pickle state.
         """
-        self.__init__(**state)
+        self.__init__(**state)  # type: ignore
 
     def __getitem__(self, index: Union[
             str, slice, int, np.integer, np.ndarray, List[int]]) -> 'Batch':
```

Comment:
It returns not only Batch (should be `Union['Batch', np.ndarray, torch.Tensor]`, same as `_parse_value`), but change it to real return type will cause other typing errors.